### PR TITLE
fix(iOS): [Commandbar] SetNavigationBar throwing exception when null

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
iOS throwing exception when setting a NavigationBar to null.

## What is the new behavior?
No exception is raised when the NavigationBar is set to null since this is commonly used in the code.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Controls/CommandBarHelper.iOS.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public static class CommandBarHelper
 	{
 		internal static void SetNavigationBar(CommandBar commandBar, UIKit.UINavigationBar? navigationBar)
 		{
-			commandBar.GetRenderer(() => new CommandBarRenderer(commandBar)).Native = navigationBar ?? throw new ArgumentNullException(nameof(navigationBar));
+			commandBar.GetRenderer(() => new CommandBarRenderer(commandBar)).Native = navigationBar;
```
Comment (jeromelaban) [inline src/Uno.UI/Controls/CommandBarHelper.iOS.cs:15]: In which conditions is this null ?

Comment (ajpinedam) [reply src/Uno.UI/Controls/CommandBarHelper.iOS.cs:15]: It is set to `null` a couple of times in the same file ex: (https://github.com/unoplatform/uno/blob/23404809f725e9d439e31e43de337364f3e7bfc5/src/Uno.UI/Controls/CommandBarHelper.iOS.cs#L104)

Comment (jeanplevesque) [reply src/Uno.UI/Controls/CommandBarHelper.iOS.cs:15]: I see it at least on lines 51, 100, 109

**Review by carldebilly — APPROVED** [review]
Change seems legit to me.

**Review by davidjohnoliver — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/beta/3.1

**Pull request merged**